npm/private/npm_translate_lock_helpers.bzl

```diff
@@ -564,6 +564,27 @@ def _normalize_bazelignore(lines):
             result.append(line.rstrip(strip_trailing_chars))
     return result
 
+################################################################################
+def _verify_lifecycle_hooks_specified(_, state):
+    # lockfiles <9.0 specify the `requiresBuild` flag in the lockfile.
+    #
+    # lockfiles >=9.0 no longer specify if packages have lifecycle hooks,
+    # and declaration of hooks must be done manually in the `pnpm.onlyBuiltDependencies`.
+    if state.lockfile_version() < 9.0:
+        return
+
+    if state.only_built_dependencies() == None:
+        fail("""\
+ERROR: in verify_lifecycle_hooks_specified:
```

Comment:
"verify_lifecycle_hooks_specified" won't be very helpful for users. Lets just summarize:

```
ERROR: pnpm.onlyBuiltDependencies required in pnpm workspace root package.json when using pnpm v9 or later
```

Reply:
For some reason all the messages in this file start with `ERROR: method_name` 🤷

Reply:
🤷